llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -1082,8 +1088,52 @@ bool X86RegisterInfo::getRegAllocationHints(Register VirtReg,
       VirtReg, Order, Hints, MF, VRM, Matrix);
 
   unsigned ID = RC.getID();
-  if (ID != X86::TILERegClassID)
+
+  if (!VRM)
+    return BaseImplRetVal;
+
+  if (ID != X86::TILERegClassID) {
```

Comment:
In order to not introduce compiler-time regression. We should use

```
if (DisableRegAllocNDDHints || !MF.getSubtarget<X86Subtarget>().hasNDD() || !isGPR(RC))
```

See X86DomainReassignment.cpp for the def of isGPR

Reply:
[4ce24cc](https://github.com/llvm/llvm-project/pull/98603/commits/4ce24cc7e550e34c28f31c046241f857ce5a03ba)